index.js

```diff
@@ -27,13 +27,20 @@ SASSPlugin.prototype.toTree = function(tree, inputPath, outputPath, inputOptions
     }
     var output = options.outputPaths[file];
     if (input) {
+      // the includePaths on addon point to addon/styles, input should be relative to includePaths
+      if(isAddonTree(inputPath)) {
+        input = input.replace(tree + "/", '')
+      }
       trees.push(new SassCompiler(inputTrees, input, output, options));
     }
     return trees;
   }, []);
 
   function tryFile(file) {
-    var filePath = path.join('.', inputPath, file);
+    // When ember-cli preprocess addon style the inputPath is '/',
+    // add the '/addon/styles' to try found addon sass files
+    var root = isAddonTree(inputPath) ? tree : ".";
+    var filePath = path.join(root, inputPath, file);
```

Comment:
@dukex this line is throwing an error when I try to build an application that uses an addon with this branch:

```
TypeError: Arguments to path.join must be strings
    at Object.posix.join (path.js:488:13)
    at tryFile (node_modules/ember-cli-sass/index.js:48:25)
```

where `root` evaluated to `{ read: [Function], cleanup: [Function] }`